clang/lib/Driver/ToolChains/SYCL.cpp

```diff
@@ -271,16 +271,20 @@ void SYCL::fpga::BackendCompiler::ConstructJob(Compilation &C,
   } else {
     // Output directory is based off of the first object name
     for (Arg * Cur : Args) {
-      SmallString<128> AN = Cur->getSpelling();
-      StringRef Ext(llvm::sys::path::extension(AN));
-      if (!Ext.empty()) {
-        types::ID Ty = getToolChain().LookupTypeForExtension(Ext.drop_front());
-        if (Ty == types::TY_INVALID)
-          continue;
-        if (types::isSrcFile(Ty) || Ty == types::TY_Object) {
-          llvm::sys::path::replace_extension(AN, "prj");
-          ReportOptArg += Args.MakeArgString(AN);
-          break;
+      if (Cur->getOption().getKind() == Option::InputClass) {
+        SmallString<128> ArgName = Cur->getSpelling();
+        StringRef Ext(llvm::sys::path::extension(ArgName));
+        if (!Ext.empty()) {
+          types::ID Ty =
+              getToolChain().LookupTypeForExtension(Ext.drop_front());
+          if (types::isSrcFile(Ty) || Ty == types::TY_Object) {
+            // Project report should be saved into CWD, so strip off any
+            // directory information if provided with the input file.
+            ArgName = llvm::sys::path::filename(ArgName);
+            llvm::sys::path::replace_extension(ArgName, "prj");
+            ReportOptArg += Args.MakeArgString(ArgName);
+            break;
+          }
         }
       }
```

Comment:
This is the "early continue" suggestion from the remaining thread that I'm talking about:
```suggestion
      if (Cur->getOption().getKind() != Option::InputClass)
        continue;
      SmallString<128> ArgName = Cur->getSpelling();
      StringRef Ext(llvm::sys::path::extension(ArgName));
      if (Ext.empty())
        continue;
      types::ID Ty =
          getToolChain().LookupTypeForExtension(Ext.drop_front());
      if (types::isSrcFile(Ty) || Ty == types::TY_Object) {   // Could also use early continue, but feels less readable smh
        // Project report should be saved into CWD, so strip off any
        // directory information if provided with the input file.
        ArgName = llvm::sys::path::filename(ArgName);
        llvm::sys::path::replace_extension(ArgName, "prj");
        ReportOptArg += Args.MakeArgString(ArgName);
        break;
      }
```

Reply:
Thanks for the suggestion.  Also found that the `TY_INVALID` check is needed as there is no guarantee that the extension is going to provide a reasonable type that we expect.

Reply:
Fair enough! Thanks.